### ISSUE TITLE
Fix buggy Learn button/next-challenge functionality and add it back in

### ISSUE THREAD

**Issue by QuincyLarson** [FreeCodeCamp/FreeCodeCamp#1778]

**Comment by QuincyLarson** [comment]
Note that some challenges are sending campers to entirely different areas of the curriculum. Specifically, from early HTML challenges to JS challenges.

**Comment by labiej** [comment]
I have sometimes experienced this when I opened a previous bonfire to help fellow campers.
When I would finish my own waypoint it would continue from the next challenge after the challenge I opened last. 

I confirmed it once more by the following steps

1. Open `Waypoint: Set the Font Family of an Element`
2. Open `Waypoint: Build Web Apps with Expressjs`
3. Finish `Waypoint: Set the Font Family of an Element`
4. You end up at `Waypoint: Store Data in MongoDB` 

Don't know if this is/was known or not.
